homeassistant/config.py

```diff
@@ -1183,36 +1374,6 @@ async def async_check_ha_config_file(hass: HomeAssistant) -> str | None:
     return res.error_str
 
 
-@callback
-def async_notify_setup_error(
```

Comment:
This was previously called via `async_log_schema_error` and `async_log_config_validator_error` when processing the configuration.
Processing the configuration no longer calls `async_log_schema_error` and `async_log_config_validator_error` though, so.
- Do we no longer get persistent notifications during startup of Home Assistant when setup calls `async_process_component_and_handle_errors` ?
- I think what needs to happen is that setup.py needs to call `async_process_component` instead, and create the needed notifications.
- There are other callers to `async_log_schema_error` and `async_log_config_validator_error` where we do not want notifications to be created. Please remove the calls to `async_notify_setup_error` from `async_log_schema_error` and `async_log_config_validator_error`

Since tests pass without these changes, I think a PR adding tests for the current notification behavior is needed and should be merged before this PR is merged.

Reply:
Tests were added and merged see: https://github.com/home-assistant/core/pull/104384

Notifications are now added via `setup.py` The split was made.